includes/class-pagespeed.php

```
@@ -354,13 +357,119 @@ private static function prepare_response( array $response ): array {
+
+			// Case 2: Singular post — try to resolve URL to post ID.
+			$post_id = url_to_postid( $url );
+			if ( $post_id > 0 ) {
+				update_post_meta( $post_id, '_wppo_lcp_image_url_' . $strategy_suffix, $lcp_url );
+				return;
+			}
+
+			// Case 3: Arbitrary URL — store in transient keyed by strategy + URL hash.
+			$transient_key = Util::transient_key( 'wppo_lcp_url_' . $strategy_suffix . '_' . md5( $normalised_scan_url ) );
+			set_transient( $transient_key, $lcp_url, DAY_IN_SECONDS );
 		}
```

PHPMD (2.15.0)

[error] The method store_lcp_image_url is not named in camelCase. (undefined)

(CamelCaseMethodName)